### PR TITLE
Migrate from flow to typescript

### LINKED ISSUE #795: Add new `jest` linting rules
https://github.com/jest-community/eslint-plugin-jest/pull/160
https://github.com/jest-community/eslint-plugin-jest/pull/164

Related: https://github.com/wemake-services/wemake-vue-template/pull/788

### LINKED ISSUE #936: Add new rules from eslint-plugin-unicorn

### LINKED ISSUE #1034: Add eslint-plugin-nuxt
https://github.com/nuxt/eslint-plugin-nuxt

### PR DESCRIPTION
**Pull request by sobolevn**
TODO:
- [x] docs rewrite
- [ ] change main tool for docs: https://github.com/wemake-services/wemake-vue-template/issues/926
- [x] implement features from opened tickets

Closes #1034
Closes #936
Closes #795

### CODE REVIEW

File: template/client/logics/api/comments.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// @flow
```
Comment (theLastOfCats) [inline template/client/logics/api/comments.ts]: Copy paste?

File: template/nuxt.config.ts [hunk 2]
```diff
@@ -52,10 +52,8 @@ module.exports = {
    * Extra Nuxt modules
    */
   'modules': [
-    'nuxt-babel',
-    'nuxt-imagemin',
+    // 'nuxt-imagemin',
```
Comment (theLastOfCats) [inline template/nuxt.config.ts]: If this module is needed, explain the purpose of commenting.
If not, then remove it.

Comment (sobolevn) [reply template/nuxt.config.ts]: Oops, yeap. This is still required. Good catch.

File: template/client/logics/api/comments.ts (hunk 1 above)
Comment (sobolevn) [reply template/client/logics/api/comments.ts]: Fixed 👍

**Pull request merged**